k8srm-prototype/pod-spec-examples.yaml

```diff
@@ -0,0 +1,133 @@
+# EXAMPLE SNIPPETS FROM PodSpec
+#
+# NOTE: We want to allow a set of "must satisfy all"
+# and multiple sets of "must satisfy one of", and
+# we want to allow matchAttributes/distinctAttributes
+# across all of the selected devices.
+#
+# As written here, each entry in deviceClaims is claimName that
+# points to an existing claim or a template with:
+# - ObjectMeta
+# - MatchAttributes (and maybe DistinctAttributes?) that apply across all
+#   claims in the Claims list
+# - Claims list, each of which is either a claim (inlined), or
+#   a "oneOf" list of claims.
+
+# This means we have an extra "  claims:" line in the simplest of cases.  We
+# could eliminate that with equivalent expressiveness, by instead listing
+# "allOf" and "oneOf", where "oneOf" is actually a list-of-lists of claims, but
+# not sure that's better.
+
+
+# Allocates four SR-IOV VFs from any Intel NIC These could even each come from
+# a different NIC (on the same node of course). NICs are not tied to specific
+# containers, so there is no need to name the claim; the NICs will be added to
+# the pod-level sandbox.
+deviceClaims:
+- template:
+    claims:
+    - deviceClass: "intel-sriov-vf"
+      requests:
+        count: 4
+
+# Allocates four SR-IOV VFs from any vendor NIC, making sure all the VFs are on
+# the same PF this requires vendor & PF together to fully qualify a physical
+# card on a node.
+deviceClaims:
+- template:
+    claims:
+    - deviceType: "sriov-vf"
+      count: 4
+      matchAttributes:
+      - "vendor"
+      - "pf"
+
+# Allocates two A100 GPUs on the same PCI root complex. The name is used to
+# attach the associated claims to a container.
+deviceClaims:
+- template:
+    claims:
+    - name: gpus
+      deviceClass: "nvidia-a100"
+      count: 2
+      matchAttributes:
+      - pcie-root
+
+# Allocates two A100 GPUs on the same PCI root complex, but keeps them
+# separated in the API so they can be attached to two different containers.
+deviceClaims:
+- template:
+    matchAttributes:
+    - pcie-root
+    claims:
+    - name: gpu-one
+      deviceClass: "nvidia-a100"
+    - name: gpu-two
+      deviceClass: "nvidia-a100"
+
+# Allocates an SR-IOV PF and a an A100 from the same pcie-root. Name is
+# populated here for the PF just to show that it can be, but the expectation is
+# generally for network devices it is not needed. Note ANY vendor PF is fine
+# here; we use deviceType, not deviceClass.
+deviceClaims:
+- template:
+    matchAttributes:
+    - pcie-root
+    claims:
+    - name: gpu
+      deviceClass: "nvidia-a100"
+    - name: nic
+      deviceType: "sriov-pf"
+
+# Allocates either an A100, H100, or two L4s that are on the same pcie root.
+deviceClaims:
+- template:
+    claims:
+    - name: gpus
+      oneOf:
+      - deviceClass: "nvidia-a100"
```

Comment:
Does this infer `count: 1` or does it mean "at least 1" ?  From discussion, it's not clear to me if depending on `limits` is semantically appropriate or not.

Reply:
I would say this implies exactly 1 (unless something else is encoded in the class (e.g. a count) that says otherwise).